lib/modules/datasource/crate/index.ts

```diff
@@ -362,6 +348,56 @@ export class CrateDatasource extends Datasource {
     return registry;
   }
 
+  private static async clone(
+    registryFetchUrl: string,
+    clonePath: string,
+    packageName: string,
+    cacheKeyForError: string,
+  ): Promise<string> {
+    const git = Git({
+      ...simpleGitConfig(),
+      maxConcurrentProcesses: 1,
+    }).env(getChildEnv());
+
+    const innerClonePromise = git.clone(registryFetchUrl, clonePath, {
+      '--depth': 1,
+    });
+    try {
+      return await innerClonePromise;
+    } catch (err) {
+      if (
+        err.message.includes(
+          'fatal: dumb http transport does not support shallow capabilities',
+        )
+      ) {
+        logger.info(
+          { packageName, registryFetchUrl },
+          'failed to shallow clone git registry, doing full clone',
+        );
+        const innerClonePromise = git.clone(registryFetchUrl, clonePath);
+        try {
+          return await innerClonePromise;
+        } catch (err) {
```

Comment:
```suggestion
    try {
      return await git.clone(registryFetchUrl, clonePath, {
        '--depth': 1,
      });
    } catch (err) {
      if (
        err.message.includes(
          'fatal: dumb http transport does not support shallow capabilities',
        )
      ) {
        logger.info(
          { packageName, registryFetchUrl },
          'failed to shallow clone git registry, doing full clone',
        );
        try {
          return await git.clone(registryFetchUrl, clonePath);
        } catch (err) {
```
@rarkins It can be simplified to this

Reply:
Yes please do that, thanks